cli/tests/test_file_track_untrack_commands.rs

```diff
@@ -128,4 +128,66 @@ fn test_untrack_sparse() {
     insta::assert_snapshot!(stdout, @r###"
     file1
     "###);
+    // Trying to manually track a file that's not included in the sparse working has
+    // no effect. TODO: At least a warning would be useful
+    let (stdout, stderr) = test_env.jj_cmd_ok(&repo_path, &["file", "track", "file2"]);
+    insta::assert_snapshot!(stdout, @"");
+    insta::assert_snapshot!(stderr, @"");
+    let stdout = test_env.jj_cmd_success(&repo_path, &["file", "list"]);
+    insta::assert_snapshot!(stdout, @r###"
+    file1
+    "###);
+}
+
+#[test]
+fn test_auto_track() {
```

Comment:
Whatever we decide the behavior should be, I think we should have a test for trying to `jj file track` an ignored file.

Reply:
How about tracking the ignore file? (and undoing it?)

Reply:
> How about tracking the ignore file? (and undoing it?)

Are you asking about what happens when the `.gitignore` file itself is untracked?

I'm not sure whether we have tests about that, but that question is relevant independently of this PR. What Git does, what I believe to be the correct behavior, and what I *believe* jj does, is to respect `.gitignore` regardless of whether it is tracked or not. This allows me, for example, to have a `.vscode/.gitignore` file in my `jj` repo that contains `*`, thus preventing itself and the entire `.vscode` dir from being tracked. I learned this trick from `jj` itself, which creates a `.jj/.gitignore` file in co-located dirs for the same reason.